src/nwp_consumer/internal/inputs/noaa/aws.py

```diff
@@ -0,0 +1,312 @@
+"""Implements a client to fetch ICON data from DWD."""
+import bz2
+import datetime as dt
+import pathlib
+import re
+import typing
+import urllib.request
+
+import requests
+import structlog
+import xarray as xr
+import cfgrib
+
+from nwp_consumer import internal
+
+from ._consts import GFS_VARIABLES
+from ._models import NOAAFileInfo
+
+log = structlog.getLogger()
+
+# See https://d-nb.info/1081305452/34 for a list of ICON parameters
+PARAMETER_RENAME_MAP: dict[str, str] = {
+    "t2m_instant": internal.OCFShortName.TemperatureAGL.value,
+    "tcc": internal.OCFShortName.HighCloudCover.value,
+    "dswrf_surface_avg": internal.OCFShortName.DownwardShortWaveRadiationFlux.value,
+    "dlwrf_surface_avg": internal.OCFShortName.DownwardLongWaveRadiationFlux.value,
+    "sdwe_surface_instant": internal.OCFShortName.SnowDepthWaterEquivalent.value,
+    "r": internal.OCFShortName.RelativeHumidityAGL.value,
+    "u10_instant": internal.OCFShortName.WindUComponentAGL.value,
+    "v10_instant": internal.OCFShortName.WindVComponentAGL.value,
+}
+
+COORDINATE_ALLOW_LIST: typing.Sequence[str] = ("time", "step", "latitude", "longitude")
+
+
+class Client(internal.FetcherInterface):
+    """Implements a client to fetch ICON data from DWD."""
+
+    baseurl: str  # The base URL for the ICON model
+    model: str  # The model to fetch data for
+    parameters: list[str]  # The parameters to fetch
+    conform: bool  # Whether to rename parameters to OCF names and clear unwanted coordinates
+
+    def __init__(self, model: str, hours: int = 48, param_group: str = "default") -> None:
+        """Create a new Icon Client.
+
+        Exposes a client for ICON data from DWD that conforms to the FetcherInterface.
+
+        Args:
+            model: The model to fetch data for. Valid models are "europe" and "global".
+            param_group: The set of parameters to fetch.
+                Valid groups are "default", "full", and "basic".
+        """
+        self.baseurl = "https://noaa-gfs-bdp-pds.s3.amazonaws.com"
+
+        match (param_group, model):
+            case ("default", _):
+                self.parameters = list(PARAMETER_RENAME_MAP.keys())
+                self.conform = True
+            case ("basic", "global"):
+                self.parameters = ["t2m_instant", "tcc",]
+                self.conform = True
+            case ("full", "global"):
```

Comment:
This also can't seem to have "basic" or "full" specification for the "europe" model?

Reply:
There isn't a Europe model for GFS, its only Global. There are different models for the US, but no Europe ones

Reply:
Also a modification needed in the docstring then if poss